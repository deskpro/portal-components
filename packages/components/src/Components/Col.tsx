import * as React from 'react';

const Col = ({
  children,
  ...props
}: React.DetailedHTMLProps<React.HTMLAttributes<HTMLDivElement>, HTMLDivElement>) => <
  div {...props}>{children}</div>;

export default Col;
