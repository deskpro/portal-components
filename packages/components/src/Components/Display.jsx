import React from 'react';
import classNames from 'classnames';
import Field from './Field';

class Display extends Field {
  /* eslint-disable class-methods-use-this */
  renderField() {
    return null;
  }
  /* eslint-disable class-methods-use-this */

  renderDescription = () => {
    const { description } = this.props;

    if (description) {
      return <span className="dp-pc_description">{description}</span>;
    }
    return null;
  };

  renderLabel = () => {
    const { label } = this.props;
    const htmlFor = this.id;
    /* eslint-disable jsx-a11y/label-has-for */
    return (
      <label className="dp-pc_label" htmlFor={htmlFor}>
        {label}
      </label>
    );
    /* eslint-enable jsx-a11y/label-has-for */
  };

  render() {
    return (
      <div className={classNames('dp-pc_field', this.className)}>
        {this.type !== 'hidden' ? this.renderLabel() : null}
        {this.renderField()}
        {this.renderDescription()}
      </div>
    );
  }
}

export default Display;
