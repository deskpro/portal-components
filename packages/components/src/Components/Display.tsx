import DOMPurify from 'dompurify/dist/purify.es';
import * as React from 'react';
import classNames from 'classnames';
import Field, { FieldProps } from './Field';

interface DisplayProps extends FieldProps {
  html: string;
}

class Display extends Field<DisplayProps> {
  renderField() {
    return <div dangerouslySetInnerHTML={{ __html: DOMPurify.sanitize(this.props.html, { ADD_ATTR: ['target'] }) }} />;
  }

  componentDidMount(): void {
    DOMPurify.addHook('afterSanitizeAttributes', function (node) {
      // set all elements owning target to target=_blank
      if ('target' in node) {
        node.setAttribute('target', '_blank');
        node.setAttribute('rel', 'noopener');
      }
    });
  }

  renderDescription = () => {
    const { description } = this.props;

    if (description) {
      return <span className="dp-pc_description">{description}</span>;
    }
    return null;
  };

  renderLabel = () => {
    const { label } = this.props;
    const htmlFor = this.id;
    return (
      <label className="dp-pc_label" htmlFor={htmlFor}>
        {label}
      </label>
    );
  };

  render() {
    return (
      <div className={classNames('dp-pc_field', this.className)}>
        {this.type !== 'hidden' ? this.renderLabel() : null}
        {this.renderDescription()}
        {this.renderField()}
      </div>
    );
  }
}

export default Display;
