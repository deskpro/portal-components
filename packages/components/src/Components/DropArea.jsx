import React, { Fragment, PureComponent } from 'react';
import PropTypes from 'prop-types';
import { deepMerge } from '@deskpro/js-utils/dist/objects';
import { Progress } from '../index';
import DropzoneImage from '../assets/dropzone-image.svg';
import ExclamationTriangle from '../assets/exclamation-triangle.svg';

const I18N = {
  uploading:       'Uploading',
  dragAndDropHere: 'Drag and drop file here',
  ok:              'OK',
};

class DropArea extends PureComponent {
  static propTypes = {
    progress:     PropTypes.number,
    isDragActive: PropTypes.bool.isRequired,
    i18n:         PropTypes.object,
    error:        PropTypes.object,
    clearError:   PropTypes.func,
  };

  static defaultProps = {
    progress: -1,
    i18n:     {},
    clearError() {},
    error:    null,
  };

  constructor(props) {
    super(props);

    this.i18n = deepMerge(I18N, props.i18n);
  }

  renderDrop = () => (
    <div>
      <DropzoneImage />
      <p>
        {this.i18n.dragAndDropHere}
      </p>
    </div>
  );

  renderProgress = () => {
    const { progress } = this.props;
    return (
      <div>
        {progress > 0 ? <span>{Math.round(progress)} %</span> : null}
        <Progress percent={progress} />
        <p>{this.i18n.uploading}...</p>
      </div>
    );
  };

  renderError = () => {
    const { error, clearError } = this.props;
    return (
      <div className="dp-pc_full-dnd__droparea_error">
        <ExclamationTriangle />
        <p>{error.message}</p>
        <button className="dpmsg-Button Button-large Button--danger" onClick={clearError}>
          {this.i18n.ok}
        </button>
      </div>
    );
  }

  render() {
    const {
      isDragActive,
      progress,
      error
    } = this.props;

    let content;
    if (error !== null) {
      content = this.renderError();
    } else if (progress < 0) {
      content = this.renderDrop();
    } else {
      content = this.renderProgress();
    }
    return (
      <Fragment>
        <div
          className="dp-pc_full-dnd__droparea"
          style={{
            display: isDragActive || progress !== -1 || error !== null ? 'block' : 'none'
          }}
        >
          {content}
        </div>
      </Fragment>
    );
  }
}

export default DropArea;
