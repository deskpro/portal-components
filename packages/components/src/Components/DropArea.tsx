import * as React from 'react';
import { PureComponent } from 'react';
import { deepMerge } from '@deskpro/js-utils/dist/objects';
import Progress from '../Components/Elements/Progress';
import DropzoneImage from '../tsx-assets/DropzoneImage';
import ExclamationTriangle from '../tsx-assets/ExclamationTriangle';
import type { Error } from '../types/error';
import type { I18nType } from "../types/i18n";

const I18N = {
  uploading:       'Uploading',
  dragAndDropHere: 'Drag and drop file here',
  ok:              'OK',
};

interface DropAreaProps {
  progress?:    number;
  isDragActive: boolean;
  i18n:         I18nType
  error:        Error;
  clearError:   () => void;
}

class DropArea extends PureComponent<DropAreaProps> {
  static defaultProps = {
    progress: -1,
    i18n:     {},
    clearError() {},
    error:    null,
  };
  private i18n: I18nType;

  constructor(props) {
    super(props);

    this.i18n = deepMerge(I18N, props.i18n);
  }

  renderDrop = () => (
    <div>
      <DropzoneImage />
      <p>
        {this.i18n.dragAndDropHere}
      </p>
    </div>
  );

  renderProgress = () => {
    const { progress } = this.props;
    return (
      <div>
        {progress > 0 ? <span>{Math.round(progress)} %</span> : null}
        <Progress percent={progress} />
        <p>{this.i18n.uploading}...</p>
      </div>
    );
  };

  renderError = () => {
    const { error, clearError } = this.props;
    return (
      <div className="dp-pc_full-dnd__droparea_error">
        <ExclamationTriangle />
        <p>{error.message}</p>
        <button className="dpmsg-Button Button-large Button--danger" onClick={clearError}>
          {this.i18n.ok}
        </button>
      </div>
    );
  }

  render() {
    const {
      isDragActive,
      progress,
      error
    } = this.props;

    let content;
    if (error !== null) {
      content = this.renderError();
    } else if (progress < 0) {
      content = this.renderDrop();
    } else {
      content = this.renderProgress();
    }
    return (
      <div
        className="dp-pc_full-dnd__droparea"
        style={{
          display: isDragActive || progress !== -1 || error !== null ? 'block' : 'none'
        }}
      >
        {content}
      </div>
    );
  }
}

export default DropArea;
