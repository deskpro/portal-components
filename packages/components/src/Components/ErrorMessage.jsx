import React from 'react';
import PropTypes from 'prop-types';
import { getIn } from 'formik';

const ErrorMessage = ({ name, form }) => {
  const error = getIn(form.errors, name);
  const touch = getIn(form.touched, name);

  return touch && error ? <span className="dp-pc_error_message">{error}</span> : null;
};

ErrorMessage.propTypes = {
  name: PropTypes.string.isRequired,
  form: PropTypes.object.isRequired,
};

export default ErrorMessage;
