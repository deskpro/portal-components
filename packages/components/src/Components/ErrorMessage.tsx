import * as React from 'react';
import { FormikProps, getIn } from 'formik';

interface ErrorMessageProps {
  name: string;
  touchName?: string;
  form: FormikProps<any>;
}

const ErrorMessage = ({ name, touchName = '', form }: ErrorMessageProps) => {
  const error = getIn(form.errors, name.split('.'));
  const touch = getIn(form.touched, (touchName || name).split('.'));

  return touch && error ? <span className="dp-pc_error_message">{error}</span> : null;
};


export default ErrorMessage;
