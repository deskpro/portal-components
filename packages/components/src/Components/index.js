import Checkboxes from './Choices/Checkboxes';
import DropDown, { DropDownInput } from './Choices/DropDown';
import MultipleDropDown, { MultipleDropDownInput } from './Choices/MultipleDropDown';
import Radio from './Choices/Radio';
import WebhookDropDown from './Choices/WebhookDropDown';

import Progress from './Elements/Progress';

import Checkbox from './Inputs/Checkbox';
import DatePicker from './Inputs/DatePicker';
import DateTimePicker from './Inputs/DateTimePicker';
import Email from './Inputs/Email';
import FileUpload, { FileUploadInput } from './Inputs/FileUpload';
import FullDragDrop from './Inputs/FullDragDrop';
import LikertScale from './Inputs/LikertScale';
import Number from './Inputs/Number';
import Password from './Inputs/Password';
import Search from './Inputs/Search';
import Tel from './Inputs/Tel';
import Text from './Inputs/Text';
import Textarea from './Inputs/Textarea';
import Url from './Inputs/Url';

import Col from './Col';
import Form from './Form';
import Row from './Row';
import Submit from './Submit';
import TicketForm from './TicketForm';

import FieldLayout from './FieldLayout';

export {
  Checkboxes,
  DropDown,
  DropDownInput,
  MultipleDropDown,
  MultipleDropDownInput,
  Radio,
  WebhookDropDown,

  Progress,

  Checkbox,
  DatePicker,
  DateTimePicker,
  Email,
  FileUpload,
  FileUploadInput,
  FullDragDrop,
  LikertScale,
  Number,
  Password,
  Search,
  Tel,
  Text,
  Textarea,
  Url,

  Col,
  Form,
  Row,
  Submit,
  TicketForm,

  FieldLayout
};

