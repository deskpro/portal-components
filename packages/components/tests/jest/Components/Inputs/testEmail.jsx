import React from 'react';
import renderer from 'react-test-renderer';
import { withFormik } from 'formik';
import { Form, Email } from 'Components';

describe('>>> Email --- Snapshot', () => {
  it('+++capturing Snapshot of Email', () => {
    const App = () => (
      <Form>
        <Email name="email" />
      </Form>
    );
    const FormikApp = withFormik({
      mapPropsToValues({
        email
      }) {
        return {
          email: email || 'test@deskpro.com'
        };
      },
      handleSubmit(values, { setSubmitting }) {
        setSubmitting(false);
      }
    })(App);
    const renderedValue = renderer.create(<FormikApp />).toJSON();
    expect(renderedValue).toMatchSnapshot();
  });
});
