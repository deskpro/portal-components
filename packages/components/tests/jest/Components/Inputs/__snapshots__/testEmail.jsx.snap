// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`>>> Email --- Snapshot +++capturing Snapshot of Email 1`] = `
<form
  className="dp-pc_form show-hover"
  onReset={[Function]}
  onSubmit={[Function]}
>
  <div
    className="dp-pc_field"
  >
    <input
      className="dp-pc_input"
      description=""
      id={null}
      label=""
      name="email"
      onBlur={[Function]}
      onChange={[Function]}
      type="email"
      value="test@deskpro.com"
    />
    <span
      className="indicator"
    />
    <label
      className="dp-pc_label"
      htmlFor="field_1"
    >
      
    </label>
    <div
      className="divider"
    />
  </div>
</form>
`;
